lib/rollup-funnel.js

```diff
@@ -45,25 +47,95 @@ function RollupFunnel(inputNodes, options) {
     throw new Error('Must specify exactly one of `include` or `exclude`.');
   }
 
-  this.originalInput = inputNodes;
   options = options || {};
+
+  this._depGraph = null;
+  this._depGraphTree = null;
+
   Plugin.call(this, [inputNodes], {
-    annotation: options.annotation
+    annotation: options.annotation,
+    persistentOutput: true
   });
   this.options = options;
 }
 
 RollupFunnel.prototype._copy = Funnel.prototype._copy;
 
+/**
+ * Constructs an FSTree from the paths stored in _depGraph.
+ */
+RollupFunnel.prototype._getFSTree = function(paths) {
+  var inputPath = this.inputPaths[0];
+  var entries = paths.map(function(entryPath) {
+    var absolutePath = path.join(inputPath, entryPath);
+
+    if (!fs.existsSync(absolutePath)) {
+      return;
+    }
+
+    var stat = fs.lstatSync(absolutePath);
+    var isDirectory = stat.isDirectory();
+
+    return {
+      relativePath: entryPath,
+      mode: stat.mode,
+      size: stat.size,
+      mtime: stat.mtime,
+      isDirectory: function() {
+        return isDirectory;
+      }
+    };
+  }).filter(Boolean);
+
+  return FSTree.fromEntries(entries);
+};
+
 RollupFunnel.prototype.build = function() {
   var base = this.inputPaths[0];
+
+  // Check for changes in the files included in the rollup
+  if (this._depGraph) {
+    var incomingDepGraphTree = this._getFSTree(this._depGraph);
+    var depGraphPatches = this._depGraphTree.calculatePatch(incomingDepGraphTree);
+    var hasDepGraphChanges = depGraphPatches.length !== 0;
+
+    var incomingNonDepGraphTree = this._getFSTree(this._nonDepGraph);
+    var nonDepGraphPatches = this._nonDepGraphTree.calculatePatch(incomingNonDepGraphTree);
+    var hasNonDepGraphChanges = nonDepGraphPatches.length !== 0;
+
+    if (!hasDepGraphChanges) {
+      if (!hasNonDepGraphChanges) {
+        return;
+      }
+
+      if (this.options.include) {
+        return;
+      }
+
+      if (this.options.exclude) {
+        nonDepGraphPatches.forEach(function(patch) {
+          var op = patch[0];
+          var module = patch[1];
+          var inputPath = path.join(this.inputPaths[0], module);
+          var outputPath = path.join(this.outputPath, module);
+          if (op === 'unlink') {
+            fs.unlinkSync(outputPath);
+          } else {
+            this._copy(inputPath, outputPath);
+          }
+        }, this);
+        return;
+      }
+    }
+  }
+
   var modules = [];
 
-  var entryExists = existsSync(path.join(this.inputPaths[0], this.options.rollup.entry));
+  var entryExists = existsSync(path.join(base, this.options.rollup.entry));
```

Comment:
Not intentional, just due to me overlooking it. I think I added all the `fs.existsSync`, so I'll update to use `exists-sync` instead.